system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -340,9 +340,33 @@ class CLdapConfig : implements ILdapConfig, public CInterface
         for (int numHosts=0; numHosts < getHostCount(); numHosts++)
         {
             getLdapHost(hostbuf);
+            unsigned port = strieq("ldaps",m_protocol) ? m_ldap_secure_port : m_ldapport;
+            StringBuffer sysUserDN, decPwd;
+
+            {
+                StringBuffer pwd;
+                cfg->getProp(".//@systemPassword", pwd);
```

Comment:
Seems a bit odd to be using // here - implies you don't know exactly where in the tree the value is going to be found. What happens if there is more than one?

Reply:
I followed the pattern established in the rest of this method, approximately 28 other instances of the same .//@